File: se321/se321-pz-nikola_tasic_3698/src/main/java/com/example/backend/service/PostService.java
package com.example.backend.service;

import com.example.backend.data.search.RequestParamQuery;
import com.example.backend.entity.Post;
import com.example.backend.entity.Tag;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;

import java.util.List;

public interface PostService {

	List<Post> findAll();

	List<Post> findAll(RequestParamQuery query, Pageable pageable);

	Long count(Specification<Post> query);

	Post save(Post post);

	Post update(Post post);

	Post findById(Integer postId);

	Post findBySlug(String slug);

	void deleteById(Integer postId);

	List<Tag> findAllTagsById(Integer postId);

}

File: cs322/cs322-pz-nikola_tasic_3698/blog-backend/Data/Service/PostService.cs
using blog_backend.Util;

namespace blog_backend.Data.Repository; 

public interface PostService {
	PaginatedList<Post> GetAll(Pageable? pageable);
	Post GetById(int id);
	Post GetBySlug(string slug);
	Post Save(Post post);
	Post Update(Post post);

	void Delete(int postId);
}